docs/source/gateway.rst

```diff
@@ -0,0 +1,745 @@
+a unified endpoint to handle all language model requests.
+
+A major advantage of using the `MLflow Gateway` service is its centralized management of API keys.
+By storing these keys in one secure location, organizations can significantly enhance their
+security posture by minimizing the exposure of sensitive API keys throughout the system. It also
+helps to prevent exposing these keys within code or having end-users have to manage keys safely.
+
+The gateway is designed to be flexible and adaptable, capable of hot-swapping routes if the
+configuration file at the target location is overwritten. This enables the easy incorporation
+of new providers or provider model types into the system without necessitating changes to
+applications that interface with the gateway. This level of adaptability makes the `MLflow Gateway`
+Service an invaluable tool in environments that require agility and quick response to changes.
+
+This simplification and centralization of language model interactions, coupled with the added
+layer of security for API key management, make the `MLflow Gateway` service an ideal choice for
+businesses that utilize language models on a regular basis.
+
+.. contents:: Table of Contents
+  :local:
+  :depth: 2
+
+.. _gateway-concepts:
+
+Concepts
+========
+
+Gateway Configuration
+---------------------
+
+The `MLflow Gateway` service relies on a user-provided configuration file, written in YAML,
+that defines the routes and providers available to the service. The configuration file dictates
+how the gateway interacts with various language model providers and determines the end-points that
+users can access.
+
+The configuration file includes a series of sections, each representing a unique route.
+Each route section has a name, a type, and a model specification, which includes the model
+provider, name, and configuration details. The configuration section typically contains the base
+URL for the API and an environment variable for the API key.
+
+Here is an example of a simple single-route configuration:
+
+.. code-block:: yaml
+
+    routes:
+      - name: chat
+        type: llm/v1/chat
+        model:
+          provider: openai
+          name: gpt-3.5-turbo
+          config:
+            openai_api_base: https://api.openai.com/v1
+            openai_api_key: $OPENAI_API_KEY
+
+
+In this example, we define a route named ``chat`` that corresponds to the ``llm/v1/chat`` type, which
+will use the ``gpt-3.5-turbo`` model from `OpenAI` to return query responses from the `OpenAI` service.
+
+The beauty of the Gateway Configuration lies in its flexibility and ease of use.
+If the configuration file is updated at the target location, the `Mlflow Gateway` service can hot-swap
+routes, allowing new providers or model types to be incorporated without requiring changes to
+the applications interfacing with the gateway. The hot-swap functionality is autonomous and
```

Comment:
I also don't think we should be using the "hot-swap" terminology. I think it's best to tell users that they can easily update the configuration to add or remove routes without any downtime.